python/beeai_framework/emitter/types.py

```diff
@@ -30,8 +30,10 @@ class EmitterInput(BaseModel):
     trace: InstanceOf[EventTrace] | None = None
 
 
-class EmitterOptions(BaseModel, frozen=True):
+class EmitterOptions(BaseModel):
     is_blocking: bool | None = None
     once: bool | None = None
     persistent: bool | None = None
     match_nested: bool | None = None
+
+    model_config = ConfigDict(frozen=True)
```

Comment:
could be extracted to another PR
cc @ajbozarth

Reply:
This was what I was trying to do, I can create a separate PR with this and some other small fixes if you don't want it blocked on this PR